src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java

```diff
@@ -114,7 +115,7 @@ public static Expression createWhereFilterForEntity(Expression baseWhereCriteria
     if (baseWhereCriteria == null) {
       baseWhereCriteria = typeSelector;
     } else {
-      baseWhereCriteria = baseWhereCriteria.and(typeSelector);
+      baseWhereCriteria = x("(" + baseWhereCriteria.toString() + ")").and(typeSelector);
```

Comment:
no better way of doing this, maybe `Expression.prefix/suffix/infix` could be made public in a future SDK release?
you could add a test that uses a `baseWhereCriteria` to `N1qlUtilsTest`:

```java
  @Test
  public void testCreateWhereFilterForEntityWithBaseCriteria() throws Exception {
    String expected = "(field >= 30 OR field2 = \"foo\") AND `_class` = \"java.lang.String\"";
    CouchbaseConverter converter = mock(CouchbaseConverter.class);
    when(converter.getTypeKey()).thenReturn("_class");
    EntityMetadata metadata = mock(EntityMetadata.class);
    when(metadata.getJavaType()).thenReturn(String.class);

    String real = N1qlUtils.createWhereFilterForEntity(
            x("field").gte(30).or(x("field2").eq(s("foo"))),
            converter, metadata).toString();

    assertEquals(expected, real);
  }
```